src/Phinx/Db/Adapter/SqlServerAdapter.php

```diff
@@ -1262,24 +1262,30 @@ protected function getColumnSqlDefinition(Column $column, $create = true)
      *
      * @param \Phinx\Db\Table\Index $index Index
      * @param string $tableName Table name
-     *
      * @return string
      */
     protected function getIndexSqlDefinition(Index $index, $tableName)
     {
+        $columnNames = $index->getColumns();
         if (is_string($index->getName())) {
             $indexName = $index->getName();
         } else {
-            $columnNames = $index->getColumns();
             $indexName = sprintf('%s_%s', $tableName, implode('_', $columnNames));
         }
+        $order = $index->getOrder();
+        if (!empty($order)) {
+            foreach ($order as $key => $value) {
+                    $loc = array_search($key, $columnNames);
+                    $columnNames[$loc] = sprintf('[%s] %s', $key, $value);
+            }
+        }
```

Comment:
```suggestion
        $order = $index->getOrder() ?? [];
        $columnNames = array_map(function($columnName) {
            $ret = '[' . $columnName . ']';
            if (isset($order[$columnName])) {
                $ret .= ' ' . $order[$columnName];
            }
            return $ret;
        }, $columnNames);
```

This I think keeps your desired behavior, while retaining existing behavior when not using an order with regards to placing `[` and `]` around each column name.

Reply:
@JRDuncan ack, sorry, I realize now I forgot to include `use ($order)` in the anonymous function definition when I wrote this out.

Reply:
Thanks @MasterOdin  When I pulled It I realized it wasn't working.